nisystemlink/clients/assetmanagement/models/_asset.py

```diff
@@ -0,0 +1,112 @@
+from typing import Dict, List, Optional
+
+from nisystemlink.clients.core._uplink._json_model import JsonModel
+
+from ._asset_calibration import (
+    CalibrationStatus,
+    ExternalCalibration,
+    SelfCalibration,
+    TemperatureSensor,
+)
+from ._asset_location import AssetLocation
+from ._asset_types import AssetBusType, AssetDiscoveryType, AssetType
+
+
+class Asset(JsonModel):
+    """Model for an object describing an asset with all of its properties."""
+
+    model_name: Optional[str] = None
+    """Gets or sets model name of the asset."""
+
+    model_number: int
```

Comment:
We are supporting projections, so all the query fields need to be optional. This needs to be applied down the model as well.
```suggestion
    model_number: Optional[int] = None
```
We don't need to apply this within the sub-models of the query response, like for location details, just at the top-level fields (what we will explicitly support in the projection enum, see my other comment).

Reply:
Updated as suggested